src/pudl/transform/ferc1.py

```diff
@@ -311,6 +362,60 @@ def transform_end(self, df: pd.DataFrame) -> pd.DataFrame:
         """Enforce the database schema and remove any cached dataframes."""
         return self.enforce_schema(df)
 
+    def normalize_metadata_xbrl(
+        self, xbrl_fact_names: list[str] | None
+    ) -> pd.DataFrame:
+        """Normalize XBRL metadata, select table-specific rows, and transform them.
+
+        In order to select the relevant rows from the normalized metadata, this function
+        needs to know which XBRL facts pertain to the table being transformed. These are
+        the names of the data columns in the raw XBRL data. To avoid creating a separate
+        dependency on the FERC 1 XBRL DB, we defer the assignment of this class
+        attribute until :meth:`Ferc1AbstractTableTransformer.process_xbrl` is called,
+        and read the column labels from the input dataframes.
+        """
+        # If the table has no XBRL metadata, return immediately:
+        if not self.xbrl_metadata_json:
+            return pd.DataFrame()
+
+        normed_meta = (
+            pd.json_normalize(self.xbrl_metadata_json)
+            .rename(
+                columns={
+                    "name": "xbrl_fact_name",
+                    "references.Account": "ferc_account",
+                }
+            )
+            .loc[
+                :,
+                [
+                    "xbrl_fact_name",
+                    "balance",
+                    "calculations",
+                    "ferc_account",
+                ],
+            ]
+        )
+        normed_meta = normed_meta.mask(
+            # Replace empty lists with NA
+            normed_meta.applymap(type).eq(list)
+            & ~normed_meta.astype(bool)
+        )
```

Comment:
what?

Reply:
normed_meta = normed_meta.mask(
```suggestion
        normed_meta = normed_meta[
            # Replace empty lists with NA
            normed_meta["calculations"].applymap(type).eq(list)
            & ~normed_meta["calculations"].astype(bool)
        ]
```
            # Replace empty lists with NA
            normed_meta.applymap(type).eq(list)
            & ~normed_meta.astype(bool)
        )

Reply:
I figured out a much simpler way to do this -- just treat the calculations column as a bool for masking purposes. Which meant I didn't need to replace the empty lists at all, since I just needed to mask based on their value elsewhere. So I removed this bit of code entirely. Now every calculation is a list, but some of them are empty, which will be easier to work with programmatically later.